### PR TITLE
doc: Update the Hubble section of getting started guides

### LINKED ISSUE #11151: doc: Update Install Hubble Section
Now Hubble runs embedded inside the Cilium process, so sections like https://docs.cilium.io/en/latest/gettingstarted/k8s-install-gke/#install-hubble need to be updated.

- document hubble-related options to enable hubble server and the CLI daemonset in https://docs.cilium.io/en/latest/gettingstarted/k8s-install-gke/#deploy-cilium
- document how to deploy hubble-ui. the helm template for hubble-ui might need to be updated as well.

### PR DESCRIPTION
**Pull request by michi-covalent**
- Update instructions to enable Hubble.
- Update minikube and kind guides to use experimental-install.yaml
  to deploy Cilium instead of quick-install.yaml so that Hubble is
  enabled.
- Delete outdated {hubble-install,getting-started-next-steps}.rst.
- Set SKIP_LINT for html-netlify target. There is another status
  check that performs linting, and this lets us ignore inconsistent
  spell results for now.

Fixes #11151

Signed-off-by: Michi Mutsuzaki <michi@isovalent.com>

### CODE REVIEW

**Comment by michi-covalent** [comment]
https://5ed83a81b9f11a0006912485--cilium-docs.netlify.app/gettingstarted/minikube.html

File: Documentation/gettingstarted/hubble-enable.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+Deploy Hubble using Helm:
+
+.. parsed-literal::
+
+   helm upgrade cilium |CHART_RELEASE| \\
+      --namespace kube-system \\
+      --set global.hubble.enabled=true \\
+      --set global.hubble.listenAddress=":4244" \\
+      --set global.hubble.metricsServer=":54321" \\
```
Comment (joestringer) [inline Documentation/gettingstarted/hubble-enable.rst]: Are these necessary? Any reason we avoid just using the defaults here?

Comment (michi-covalent) [inline Documentation/gettingstarted/hubble-enable.rst]: not sure what a good default port is here.

File: Documentation/gettingstarted/hubble-enable.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+for cloud native workloads. It is built on top of Cilium and eBPF to enable
+deep visibility into the communication and behavior of services as well as the
+networking infrastructure in a completely transparent manner.
+
+Deploy Hubble using Helm:
+
+.. parsed-literal::
+
+   helm upgrade cilium |CHART_RELEASE| \\
+      --namespace kube-system \\
```
Comment (michi-covalent) [inline Documentation/gettingstarted/hubble-enable.rst]: haven't figured out how to parameterize the namespace, for example gke uses `cilium` instead of `kube-system`.

**Review by joestringer — COMMENTED** [review]
Seems reasonable. I didn't try it out yet. One comment below.

File: Documentation/gettingstarted/hubble-enable.rst [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+To pick one Cilium instance and validate that Hubble server is properly configured to listen on
+a UNIX domain socket:
+
+.. parsed-literal::
+
+    kubectl exec -n kube-system -t ds/cilium -- hubble observe
+
+To validate that Hubble UI is properly configured, set up a port forwarding for ``hubble-ui`` service:
+
```
Comment (michi-covalent) [inline Documentation/gettingstarted/hubble-enable.rst:48]: i also wanted to add a command to validate hubble-relay, but `hubble observe --server hubble-relay:80` doesn't work from cilium pod because of its dns policy: https://cilium.slack.com/archives/C1MATJ5U5/p1591223752145300

File: Documentation/gettingstarted/hubble-enable.rst [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+   helm upgrade cilium |CHART_RELEASE| \\
+      --namespace kube-system \\
+      --set global.hubble.enabled=true \\
+      --set global.hubble.listenAddress=":4244" \\
+      --set global.hubble.metricsServer=":54321" \\
+      --set global.hubble.metrics.enabled="{dns,drop,tcp,flow,port-distribution,icmp,http}" \\
+      --set global.hubble.relay.enabled=true \\
+      --set global.hubble.ui.enabled=true
+
```
Comment (michi-covalent) [inline Documentation/gettingstarted/hubble-enable.rst:27]: we should add a warning here that this is opening up a port 4244 for hubble-relay, might need @Rolinh 's help to properly phrase the warning.

File: Documentation/gettingstarted/k8s-install-download-release.rst [hunk 5]
```diff
@@ -25,7 +25,7 @@
 
    Download the Cilium release tarball and change to the kubernetes install directory:
 
-    .. parsed-literal::
+   .. parsed-literal::
```
Comment (michi-covalent) [inline Documentation/gettingstarted/k8s-install-download-release.rst]: didn't include the description in the commit message, but this just fixes the indentation of the section below.

File: Documentation/gettingstarted/hubble-enable.rst (hunk 1 above)
Comment (michi-covalent) [reply Documentation/gettingstarted/hubble-enable.rst]: -  `--set global.hubble.listenAddress=":4244"` this one is necessary for hubble-relay because by default hubble only listens on a unix domain socket.
- `--set global.hubble.metricsServer=":54321"` i guess this is also necessary, hubble metrics is disabled by default: https://github.com/cilium/cilium/blob/8ba1f84d2ffe5d59b1ca0a94c8ab804a9d43feac/install/kubernetes/cilium/values.yaml#L463

File: Documentation/gettingstarted/hubble-enable.rst (hunk 2 above)
Comment (joestringer) [reply Documentation/gettingstarted/hubble-enable.rst]: Maybe... https://stackoverflow.com/questions/58930190/sphinx-and-restructuredtext-pass-parameter-in-include-directive ?

File: Documentation/gettingstarted/hubble-enable.rst (hunk 4 above)
Comment (joestringer) [reply Documentation/gettingstarted/hubble-enable.rst:27]: Do we expose it on addresses other than loopback by default?

File: Documentation/gettingstarted/hubble-enable.rst (hunk 2 above)
Comment (michi-covalent) [reply Documentation/gettingstarted/hubble-enable.rst]: i found the same link :) it didn't work with sphinx unfortunately https://stackoverflow.com/questions/58930190/sphinx-and-restructuredtext-pass-parameter-in-include-directive#comment104245250_58930571

File: Documentation/gettingstarted/hubble-enable.rst (hunk 4 above)
Comment (michi-covalent) [reply Documentation/gettingstarted/hubble-enable.rst:27]: unix domain socket only by default.

with hubble-relay we need to specify `--set global.hubble.listenAddress`. the address needs to be reachable by hubble-relay, we typically do `":4244"`.

File: Documentation/gettingstarted/hubble-enable.rst (hunk 1 above)
Comment (gandro) [reply Documentation/gettingstarted/hubble-enable.rst]: I don't know either, but right now Cilium exposes metrics on the following ports:
- 9090 cilium
- 6942 cilium-operator

I think 9091 which we use in CI might be a decent choice. I wonder if it maybe makes sense to rewrite the helm charts so we always have a default port and users only need to set `global.hubble.metrics.enabled`.

File: Documentation/gettingstarted/minikube.rst [hunk 6]
```diff
@@ -56,24 +50,20 @@ Install kubectl & minikube
...
 .. note::
 
    In case of installing Cilium with CRIO, please see :ref:`crio-instructions` instructions.
 
+Install Cilium using Helm:
+
 .. parsed-literal::
 
-    kubectl create -f \ |SCM_WEB|\/install/kubernetes/quick-install.yaml
+   helm install cilium |CHART_RELEASE| --namespace kube-system
```
Comment (gandro) [inline Documentation/gettingstarted/minikube.rst]: While I fully understand why we do this, I really liked the fact that the old getting started guide in Cilium does not require users to install any external tools (like helm) unless they use more advanced features/configuration.

While I'm not fundamentally opposed to requiring helm in Getting Started, maybe providing a `quick-install-with-hubble.yaml` would work nicely as well. In local testing, it seems like we can `kubectl apply` this over an existing quick-install.

If we keep Helm as part of the getting started guide, we should probably also explain how to obtain Helm (since we already explain how to obtain kubectl and minikube there).

**Review by gandro — COMMENTED** [review]
Left a comment about the use of Helm. Otherwise this looks great from the Hubble side!

File: Documentation/gettingstarted/hubble-enable.rst (hunk 4 above)
Comment (rolinh) [reply Documentation/gettingstarted/hubble-enable.rst:27]: And what about updating the guide to do this in two (or three) steps?
What I'm thinking of is:
step 1: enable hubble
step 2: enable hubble on port 4244 + deploy hubble relay
step 2 (or 3?): deploy hubble UI

This would allow us to clearly state what you get at each step and add appropriate warnings for step 2+ so that users may stop at step 1 and still get the benefits of local hubble.

File: Documentation/gettingstarted/hubble-enable.rst (hunk 1 above)
Comment (gandro) [reply Documentation/gettingstarted/hubble-enable.rst]: PR to simplify the metric values: https://github.com/cilium/cilium/pull/11887

File: Documentation/gettingstarted/minikube.rst (hunk 6 above)
Comment (michi-covalent) [reply Documentation/gettingstarted/minikube.rst]: filed https://github.com/cilium/cilium/issues/11902

Comment (michi-covalent) [reply Documentation/gettingstarted/minikube.rst]: https://github.com/cilium/cilium/pull/11907

File: Documentation/gettingstarted/hubble-enable.rst (hunk 1 above)
Comment (michi-covalent) [reply Documentation/gettingstarted/hubble-enable.rst]: rebased to pick up #11887

File: Documentation/gettingstarted/hubble-enable.rst (hunk 3 above)
Comment (michi-covalent) [reply Documentation/gettingstarted/hubble-enable.rst:48]: https://github.com/cilium/cilium/pull/11881

File: Documentation/gettingstarted/hubble-enable.rst (hunk 2 above)
Comment (michi-covalent) [reply Documentation/gettingstarted/hubble-enable.rst]: added a step to specify the namespace as an env variable.

**Comment by michi-covalent** [comment]
blocked on https://github.com/cilium/cilium/pull/11881 and https://github.com/cilium/cilium/pull/11907

File: Documentation/gettingstarted/hubble-enable.rst [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+           helm upgrade cilium |CHART_RELEASE| \\
+              --namespace $CILIUM_NAMESPACE \\
+              --set global.hubble.enabled=true \\
+              --set global.hubble.metrics.enabled="{dns,drop,tcp,flow,port-distribution,icmp,http}"
+
+     .. group-tab:: Distributed Mode (beta)
+
+
+        In **distributed mode (beta)**, Hubble listens on a TCP port on the host network.
+        This allows Hubble Relay to communicate with all the Hubble instances in the cluster.
```
Comment (aanm) [inline Documentation/gettingstarted/hubble-enable.rst]: Probably worth adding a warning this exposes all traffic metadata insecurely in that TCP port.

**Comment by Rolinh** [comment]
> blocked on #11881 and #11907

Are you really blocked on #11881 @michi-covalent? The way I understand it, it's only a problem when one tries to reach hubble-relay when using the hubble CLI binary from the cilium pod which is not really something we want to advise doing. Instead, I think we want to promote the port-forwarding method + using released binaries of hubble CLI to query Hubble Relay.

File: Documentation/gettingstarted/hubble-enable.rst [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+* Specify the namespace in which Cilium is installed as ``CILIUM_NAMESPACE``
+  environment variable. Subsequent commands reference this environment variable.
+
+  .. parsed-literal::
+
+      export CILIUM_NAMESPACE=kube-system
+
+  .. note::
+
+      Cilium is typically installed in ``kube-system`` namespace, but there are
```
Comment (rolinh) [inline Documentation/gettingstarted/hubble-enable.rst]:
```suggestion
      Cilium is typically installed in the ``kube-system`` namespace, but there are
```

File: Documentation/gettingstarted/hubble-enable.rst [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+      a few exceptions such as GKE which install Cilium in a dedicated ``cilium``
```
Comment (rolinh) [inline Documentation/gettingstarted/hubble-enable.rst:19]:
```suggestion
      a few exceptions such as GKE which installs Cilium in a dedicated ``cilium``
```

File: Documentation/gettingstarted/hubble-enable.rst (hunk 7 above)
Comment (rolinh) [inline Documentation/gettingstarted/hubble-enable.rst]: We probably want to briefly introduce Hubble Relay before mentioning it.

File: Documentation/gettingstarted/hubble-enable.rst [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+
+
+        In **distributed mode (beta)**, Hubble listens on a TCP port on the host network.
+        This allows Hubble Relay to communicate with all the Hubble instances in the cluster.
+        Hubble CLI and Hubble UI in turn connect to Hubble Relay to provide cluster-wide
+        networking visibility.
+
+        .. warning::
+
+           In Distributed mode, Hubble runs a gRPC service over plan-text HTTP on the host
```
Comment (rolinh) [inline Documentation/gettingstarted/hubble-enable.rst]:
```suggestion
           In Distributed mode, Hubble runs a gRPC service over plain-text HTTP on the host
```

File: Documentation/gettingstarted/hubble-enable.rst [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+        provides networking visibility for traffic observed by the local Cilium agent.
+
+        .. parsed-literal::
+
+           helm upgrade cilium |CHART_RELEASE| \\
+              --namespace $CILIUM_NAMESPACE \\
+              --set global.hubble.enabled=true \\
+              --set global.hubble.metrics.enabled="{dns,drop,tcp,flow,port-distribution,icmp,http}"
+
+     .. group-tab:: Distributed Mode (beta)
```
Comment (rolinh) [inline Documentation/gettingstarted/hubble-enable.rst:39]: Neat!

File: Documentation/gettingstarted/hubble-enable.rst [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+
+  .. parsed-literal::
+
+      kubectl exec -n $CILIUM_NAMESPACE -t ds/cilium -- hubble observe
+
+* **(Distributed mode only)** To validate that Hubble Relay is running:
+
+  .. parsed-literal::
+
+      kubectl exec -n $CILIUM_NAMESPACE -t ds/cilium -- hubble observe --server hubble-relay:80
```
Comment (rolinh) [inline Documentation/gettingstarted/hubble-enable.rst]: We probably don't want to promote a method which requires to `kubectl exec` into a privileged container to then reach a service in the cluster.
Instead, I would suggest the approach described in #11919 where we tell users to `kubectl -n $CILIUM_NAMESPACE port-forward service/hubble-relay 4245:80` and then download and use a release binary of hubble CLI from their own laptop/PC targeting `localhost:4245`.

**Review by rolinh — CHANGES_REQUESTED** [review]
Very good overall! I have a few suggested changes however.

File: Documentation/gettingstarted/hubble-enable.rst (hunk 9 above)
Comment (michi-covalent) [reply Documentation/gettingstarted/hubble-enable.rst:19]: i googled this one :)

https://ell.stackexchange.com/questions/9684/a-few-takes-a-singular-or-plural-verb-in-present-simple-tense

File: Documentation/gettingstarted/hubble-enable.rst (hunk 1 above)
Comment (michi-covalent) [reply Documentation/gettingstarted/hubble-enable.rst]: removed `global.hubble.metricsServer`. i think we need to live with `global.hubble.listenAddress` for now.

File: Documentation/gettingstarted/hubble-enable.rst (hunk 7 above)
Comment (michi-covalent) [reply Documentation/gettingstarted/hubble-enable.rst]: made the wording stronger:

> In Distributed mode, Hubble runs a gRPC service over plain-text HTTP on the host network without any authentication/authorization. The main consequence is that anybody who can reach the Hubble gRPC service can obtain all the networking metadata from the host. It is therefore strongly discouraged to enable distributed mode in a production environment.

Comment (michi-covalent) [reply Documentation/gettingstarted/hubble-enable.rst]: added a link to https://docs.cilium.io/en/latest/hubble/#hubble-relay

File: Documentation/gettingstarted/minikube.rst [hunk 13]
```diff
@@ -67,13 +67,9 @@ Kubernetes CNI plugin.
 
 .. parsed-literal::
 
-    kubectl create -f \ |SCM_WEB|\/install/kubernetes/quick-install.yaml
+    kubectl create -f \ |SCM_WEB|\/install/kubernetes/experimental-install.yaml
```
Comment (gandro) [inline Documentation/gettingstarted/minikube.rst]: I think we should still default to `quick-install.yaml` and provide `experimental-install.yaml` as an option in a box below, explaining what it does. I think it looks a bit scary for a "Getting Started" guide to immediately recommend installing features which are explicitly marked not ready for production use.

**Review by gandro — COMMENTED** [review]
Love the distinction between "Local" and "Distributed" mode in Hubble. Nicely explained!

File: Documentation/Makefile [hunk 14]
```diff
@@ -48,7 +48,7 @@ epub latex html: builder-image copy-api
 
 html-netlify: copy-api
 	@$(ECHO_GEN)_build/$@
-	$(QUIET) ./check-build.sh html $(SPHINX_OPTS)
+	$(QUIET) SKIP_LINT=1 ./check-build.sh html $(SPHINX_OPTS)
```
Comment (rolinh) [inline Documentation/Makefile:51]: Is this change intended?

File: Documentation/gettingstarted/hubble-enable.rst [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+
+* To pick one Cilium instance and validate that Hubble is properly configured to listen on
+  a UNIX domain socket:
+
+  .. parsed-literal::
+
+      kubectl exec -n $CILIUM_NAMESPACE -t ds/cilium -- hubble observe
+
+* **(Distributed mode only)** To validate that Hubble Relay is running, download the ``hubble``
+  CLI from https://github.com/cilium/hubble/releases and install it in a directory listed in the
```
Comment (rolinh) [inline Documentation/gettingstarted/hubble-enable.rst]: nit:
```suggestion
  CLI from `the GitHub release page <https://github.com/cilium/hubble/releases>`_ and install it in a directory listed in the
```

File: Documentation/gettingstarted/hubble-enable.rst [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+  ``PATH`` environment variable. Then, set up a port forwarding for ``hubble-relay`` service and
+  run ``hubble observe`` command:
+
+  .. parsed-literal::
+
+      kubectl port-forward -n $CILIUM_NAMESPACE svc/hubble-relay 4245:80
+      hubble observe --server localhost:4245
+
```
Comment (rolinh) [inline Documentation/gettingstarted/hubble-enable.rst:87]:
```suggestion

For convenience, you may set and export the ``HUBBLE_DEFAULT_SOCKET_PATH``
environment variable:

.. code:: bash
    $ export HUBBLE_DEFAULT_SOCKET_PATH=localhost:4245
This will allow you to use ``hubbble status`` and ``hubble observe`` commands
without having to specify the server address via the ``--server`` flag.
```

File: Documentation/Makefile (hunk 14 above)
Comment (michi-covalent) [reply Documentation/Makefile:51]: yes i updated the commit message. this basically skips spell-checking for netlify build. it's already checked in another status check, and we had inconsistent results because netlify image has a different dictionary.

**Review by rolinh — APPROVED** [review]

**Comment by michi-covalent** [comment]
test-me-please

File: Documentation/gettingstarted/hubble-enable.rst [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+  without having to specify the server address via the ``--server`` flag.
+
+* **(Distributed mode only)** To validate that Hubble UI is properly configured, set up a port forwarding for
+  ``hubble-ui`` service:
+
+  .. parsed-literal::
+
+      kubectl port-forward -n $CILIUM_NAMESPACE svc/hubble-ui 12000
+
+  and then open http://localhost:12000/.
```
Comment (joestringer) [inline Documentation/gettingstarted/hubble-enable.rst:105]: Should this validation section also be split by local / distributed modes?

**Review by joestringer — APPROVED** [review]
One minor question

File: Documentation/gettingstarted/hubble-enable.rst (hunk 17 above)
Comment (michi-covalent) [reply Documentation/gettingstarted/hubble-enable.rst:105]: i kind of wanted to keep these parts visible by default so that people see what additional capabilities they get from the distributed mode.
